docs/design/architechture.md

```diff
@@ -181,63 +205,172 @@ metadata:
   name: kernel-y
   namespace: ml-apps
 spec:
-  ociImage:
-    pullPolicy: IfNotPresent
-    image: quay.io/example/kernel-y:latest
-    pullSecret:
-  validateSignature: false
+  cacheImage: quay.io/example/kernel-y:latest
 status:
   conditions:
-  - lastTransitionTime: "2025-05-08T21:06:07Z"
-    message: 'TritonKernelCache Reconciliation failed on the following TritonKernelCacheNodeStatus
-      objects: [kernel-y-node1]'
-    reason: Error
-    status: "True"
-    type: Error
+    - lastTransitionTime: "2025-05-08T21:06:07Z"
+      message: 'TritonKernelCache reconciliation failed on the following nodes: [node1]'
+      reason: Error
+      status: "True"
+      type: Error
 ```
 
-> *[OI] I don't think we need `validateSignature` here? We will want
-> a TKM configuration option to allow unsigned images and to disable
-> COSign. bpfman had a ConfigMap which contained a configuration file.*
-
 #### TritonKernelCacheNodeStatus and TritonKernelCacheNodeStatusCluster CRD
 
 TritonKernelCacheNodeStatus and TritonKernelCacheNodeStatusCluster CRD instances
 are created by the TKM Agent, not the user.
-There is an instance for each Kubernetes Node for each TritonKernelCache or
-TritonKernelCacheCluster instance.
-The purpose is to provide the status of a given Triton kernel for each node
-as well detected GPU info from the node.
+Each node will have one such CR to represent its status, and it will contain
+information for all kernel caches relevant to that node, organized per GPU.
+
+If the corresponding TritonKernelCache is namespace-scoped, the NodeStatus CR
+should live in the same namespace.If the corresponding TritonKernelCache is c
+luster-scoped, a cluster-scoped NodeStatus CR (TritonKernelCacheNodeStatusCluster)
+will be used instead.
+
+While nodes themselves are not namespaced, the namespace of the NodeStatus CR
+follows the scope of the kernel cache resource it reports on. This allows the
+operator to correctly associate status objects with their source cache definitions.
+
+This structure enables more efficient status tracking in environments with
+heterogeneous GPU configurations and supports CSI plugin queries via the TKM Agent.
+
+Summary of data reflected in the CRD:
+
+- **nodeName**: The name of the Kubernetes node.
+- **gpus**: List of physical GPUs on the node with:
```

Comment:
If we group GPUs by type then:
gpus: List of physical GPUs on the node grouped by type and driver, each with:
  - gpuType: (e.g., nvidia-a100)
  - driverVersion: The GPU driver version
  - ids: List of GPU indices (e.g., [0, 1, 2])